### DISCUSSION TITLE
Github Classroom no longer in active development ? Any alternatives ?

### DISCUSSION THREAD

**Discussion by Caesarovich** [community/community#73287] [GitHub Education] [answered]
Hi,

As of the November 2nd 2023, the Github Classroom Extension received a new release (0.0.4) with the following changes:

> 0.0.4 (November 2, 2023)
> Added a note to the README that this extension is not under active development. We still expect the one-click "Open in VS Code" Experience on GitHub Classroom to work, but some features of the extension may not work as expected in the future
> Changed the support link to https://github.com/orgs/community/discussions/categories/github-education

This makes it official that Github Classroom won't see any improvements in the future. This is a very sad news for me, because I use this tool every day with my students. 

I am now wondering if anyone could recommend me alternatives for GH Classroom. Are there even such alternatives being developed ?
It would be awesome to see Open Source alternatives coming up. Perhaps just a community made Extension for GH Classroom at the very least.

Github Education Team, if you do not plan to actively develop this extension, maybe could you open it to the community ? You guys always say how much you love Open source ! 

I leave this post as an open letter to discuss this matter and hopefuly find people to work on solutions !

**Comment by RyanHecht** [comment, marked as answer]
Hi there! I'm the Product Manager for GitHub Classroom. A point of clarification: GitHub Classroom (https://classroom.github.com, the webapp) is still being actively developed! The GitHub Classroom Visual Studio Code extension is what is no longer under active development. This extension was released in Beta in August 2021, and has not received any updates since the month it was released. As part of some internal hygiene, we decided to add this notice to the README to make it clear that we have no short-term plans to make any updates to this Visual Studio Code extension.

The good news is that if you're currently using the Visual Studio Code extension for students to browse Classroom assignments, easily sync progress back to GitHub, and see autograding test results, these features should still work barring any breaking updates to Visual Studio Code! The one-click "Open in VS Code" experience from student repositories should also continue to function as expected.

As far as open source alternatives to this extension are concerned: We recently open sourced the [GitHub Classroom Extension for the GitHub CLI](https://github.com/github/gh-classroom) and along with it the [REST API](https://docs.github.com/en/rest/classroom/classroom?apiVersion=2022-11-28) for interacting with GitHub Classroom! We'd love to see community-created Visual Studio Code extensions that implement new features that our extension doesn't currently provide!

Rest assured, we're still hard at work creating new features and functionality for the core GitHub Classroom experience -- I can't wait to share more very soon!

**Reply by Caesarovich** [reply]
Thanks for your reply @RyanHecht , I am happy to learn that the project is still actively developed. How can I stay in touch with future updates and provide feedbacks ?

**Reply by ssyberg** [reply]
🦗

**Reply by Cheplusplus** [reply]
Github classrooms is so broken its hard to believe
